tests/conftest.py

```diff
@@ -3,11 +3,13 @@
 import os
 import subprocess
 import sys
+import tempfile
 from collections import defaultdict
 from pathlib import Path
+from typing import Generator
```

Comment:
Deprecated since version 3.9: [collections.abc.Generator](https://docs.python.org/3/library/collections.abc.html#collections.abc.Generator) now supports subscripting ([]). See [PEP 585](https://peps.python.org/pep-0585/) and [Generic Alias Type](https://docs.python.org/3/library/stdtypes.html#types-genericalias)  (source: https://docs.python.org/3/library/typing.html?highlight=typing#typing.Generator)

You can also just use `Iterator` too if you don't need to specify the extra types in a `Generator`:

```python
from collections.abc import Iterator

def oh_it_iterates() -> Iterator[str]:
    yield "iter"
    yield "ator"
    yield "!!!"
```

Reply:
Oh cool! I didn't know this!

Reply:
hm since this was a change introduced in Python 3.9 I think we should wait until we drop Python 3.8 support before making the switch from `typing` to `collections.abc`